### PR TITLE
[docs] restructure nonlinear optimization tutorials

### PR DESCRIPTION
**Pull request by odow**
Related to https://github.com/jump-dev/JuMP.jl/pull/3822.

This especially fixes the issue that `User-defined operators with vector outputs` was the only subsection of the "tips and tricks" tutorial because we never got around to adding other tricks...

image

### CODE REVIEW

File: docs/src/changelog.md [hunk 1]
```diff
@@ -261,7 +261,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
     - `Percival.jl` (#3567)
  - Added new tutorials:
     - [Approximating nonlinear functions](@ref) (#3563)
-    - [Classifiers](@ref) (#3569)
+    - [Exmaple: classification problems](@ref) (#3569)
```
Comment (odow) [inline docs/src/changelog.md:264]:
```suggestion
    - [Example: classification problems](@ref) (#3569)
```

File: docs/src/tutorials/nonlinear/classifiers.jl [hunk 2]
```diff
@@ -3,7 +3,7 @@
 # v.2.0. If a copy of the MPL was not distributed with this file, You can       #src
 # obtain one at https://mozilla.org/MPL/2.0/.                                   #src
 
-# # Classifiers
+# # Exmaple: classification problems
```
Comment (odow) [inline docs/src/tutorials/nonlinear/classifiers.jl:6]:
```suggestion
# # Example: classification problems
```

**Comment by odow** [comment]
Now this is
image

The computing hessians tutorial should really become "working with the MOI evaluator".

File: docs/src/changelog.md [hunk 3]
```diff
@@ -261,7 +261,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
     - `Percival.jl` (#3567)
  - Added new tutorials:
     - [Approximating nonlinear functions](@ref) (#3563)
-    - [Classifiers](@ref) (#3569)
+    - [Example: classification problems](@ref) (#3569)
```
Comment (jd-foster) [inline docs/src/changelog.md:264]: Just checking you intended to rewrite the changelog?

Comment (odow) [reply docs/src/changelog.md:264]: Yip

Comment (jd-foster) [reply docs/src/changelog.md:264]: Ah, it needs to match the new ref.

**Pull request merged**